repo2docker/contentproviders/zenodo.py

```diff
@@ -4,78 +4,46 @@
 
 from os import makedirs
 from os import path
-from urllib.request import urlopen, Request
+from urllib.request import Request
 from urllib.error import HTTPError
-from zipfile import ZipFile, is_zipfile
 
-from .base import ContentProvider
+from .doi import DoiProvider
 from ..utils import copytree, deep_get
-from ..utils import normalize_doi, is_doi
-from .. import __version__
 
 
-class Zenodo(ContentProvider):
+class Zenodo(DoiProvider):
     """Provide contents of a Zenodo deposit."""
 
-    def _urlopen(self, req, headers=None):
-        """A urlopen() helper"""
-        # someone passed a string, not a request
-        if not isinstance(req, Request):
-            req = Request(req)
-
-        req.add_header("User-Agent", "repo2docker {}".format(__version__))
-        if headers is not None:
-            for key, value in headers.items():
-                req.add_header(key, value)
-
-        return urlopen(req)
-
-    def _doi2url(self, doi):
-        # Transform a DOI to a URL
-        # If not a doi, assume we have a URL and return
-        if is_doi(doi):
-            doi = normalize_doi(doi)
-
-            try:
-                resp = self._urlopen("https://doi.org/{}".format(doi))
-            # If the DOI doesn't resolve, just return URL
-            except HTTPError:
-                return doi
-            return resp.url
-        else:
-            # Just return what is actulally just a URL
-            return doi
+    # We need the hostname (url where records are), api url (for metadata),
+    # filepath (path to files in metadata), filename (path to filename in
+    # metadata), download (path to file download URL), and type (path to item type in metadata)
+    hosts = [
```

Comment:
Could we move this into a `__init__` method and make it define `self.hosts` instead of here?

Mostly because class attributes and such are traps people eventually fall into later in a Python project when modifying it for once instance seems to modify it for all instances "as if by magic". Hence I'd prefer setting it explicitly as a instance variable.